mirar/processors/photcal/zp_calculator/color_term_zp_calculator.py

```diff
@@ -60,9 +82,35 @@ def calculate_zeropoint(
                 matched_img_cat[colname.replace("MAG", "MAGERR")] ** 2
                 + matched_ref_cat["magnitude_err"] ** 2
             )
-            popt, pcov = curve_fit(f=line_model, xdata=x, ydata=y, sigma=y_err)
-            color, zero_point = popt
-            color_err, zp_err = np.sqrt(np.diag(pcov))
+            x_err = np.sqrt(
+                matched_ref_cat[color_err_colnames[0]] ** 2
+                + matched_ref_cat[color_err_colnames[1]] ** 2
+            )
+
+            # use scipy.odr to fit a line to data with x and y uncertainties
+            ## setup: remove sources with 0 uncertainty (or else scipy.odr won't work)
+            where_zero_y = np.where(np.array(y_err) == 0)[0]
```

Comment:
Probably cleaner with `non_zero_mask = (np.array(y_err)==0) | (np.array(x_err)==0)`

Reply:
And then `x_err = x_err[non_zero_mask]` and so on, will have to convert all lists to numpy arrays